qiskit_experiments/library/hamiltonian/heat_base.py

```diff
@@ -0,0 +1,243 @@
+        the entangler pulse sequence of interest, thus one must provide the definition of it
+        through the backend or custom transpiler configuration, i.e. instruction schedule map.
+        This gate name can be overridden via the experiment option of this experiment.
+
+    # section: note
+
+        This class is usually not exposed to end users.
+        Developer of new HEAT experiment must design amplification sequence and
+        instantiate the class implicitly in the batch experiment.
+        The :class:`BatchHeatHelper` provides a convenient wrapper class of
+        the :class:`qiskit_experiments.framework.BatchExperiment` for implementing a
+        typical HEAT experiment.
+
+    # section: analysis_ref
+        :py:class:`HeatElementAnalysis`
+
+    # section: reference
+        .. ref_arxiv:: 1 2007.02925
+    """
+
+    def __init__(
+        self,
+        qubits: Tuple[int, int],
+        prep_circ: QuantumCircuit,
+        echo_circ: QuantumCircuit,
+        meas_circ: QuantumCircuit,
+        backend: Optional[Backend] = None,
+        parameter_name: Optional[str] = "d_theta",
+        **kwargs,
+    ):
+        """Create new HEAT sub experiment.
+
+        Args:
+            qubits: Index of control and target qubit, respectively.
+            prep_circ: A circuit to prepare qubit before the echo sequence.
+            echo_circ: A circuit to selectively amplify the specific error term.
+            meas_circ: A circuit to project target qubit onto the basis of interest.
+            backend: Optional, the backend to run the experiment on.
+            parameter_name: A name that represents angle from fitting.
+
+        Keyword Args:
+            See :meth:`experiment_options` for details.
+        """
+        analysis = HeatElementAnalysis()
+        analysis.set_options(result_parameters=[ParameterRepr("d_theta", parameter_name, "rad")])
+
+        super().__init__(qubits=qubits, backend=backend, analysis=analysis)
+        self.set_experiment_options(**kwargs)
+
+        # These are not user configurable options. Be frozen once assigned.
+        self._prep_circuit = prep_circ
+        self._echo_circuit = echo_circ
+        self._meas_circuit = meas_circ
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            repetitions (Sequence[int]): A list of the number of echo repetitions.
+            cr_gate (Gate): A gate instance representing the entangler sequence.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(21))
+        options.heat_gate = circuit.Gate("heat", num_qubits=2, params=[])
+
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpile options."""
+        options = super()._default_transpile_options()
+        options.basis_gates = ["sx", "x", "rz", "heat"]
```

Comment:
This exist both in the child and batch class. This is because the option set to the batch class should be propagated. For example
```python
exp = HeatZX(...)
exp.set_transpile_options(some_opt=some_value)
```
then this value is copied to each child classes.

Reply:
Ok. Unrelated to this PR but looking at this it feels like we are hard-coding the basis_gates, I know we have done this in other experiments (myself included). Perhaps it might be worth having a mechanism where we augment the basis gate set of the backend. In this case we would add "heat". This would however be another PR.

Reply:
something like extended_basis_gate? That sounds good to me. Of course user can update since this is just an option. In the unittest I replaced "heat" with "unitary" so that AerSimulator can simulate it with attached matrix definition.